File: src/Microsoft.Management.Deployment/PackageAgreement.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
#include "pch.h"
#include "PackageAgreement.h"
#include "PackageAgreement.g.cpp"
#include <wil\cppwinrt_wrl.h>

namespace winrt::Microsoft::Management::Deployment::implementation
{
    void PackageAgreement::Initialize(::AppInstaller::Manifest::Agreement packageAgreement)
    {
        m_packageAgreement = std::move(packageAgreement);
    }
    hstring PackageAgreement::Label()
    {
        return winrt::to_hstring(m_packageAgreement.Label);
    }
    hstring PackageAgreement::Text()
    {
        return winrt::to_hstring(m_packageAgreement.AgreementText);
    }
    hstring PackageAgreement::Url()
    {
        return winrt::to_hstring(m_packageAgreement.AgreementUrl);
    }
}

File: src/WinGetUtilInterop/Manifest/V1/PackageAgreement.cs
// -----------------------------------------------------------------------------
// <copyright file="PackageAgreement.cs" company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
// </copyright>
// -----------------------------------------------------------------------------

namespace Microsoft.WinGetUtil.Models.V1
{
    /// <summary>
    /// Class that contains package agreement.
    /// </summary>
    public class PackageAgreement
    {
        /// <summary>
        /// Gets or sets the agreement label.
        /// </summary>
        public string AgreementLabel { get; set; }

        /// <summary>
        /// Gets or sets the agreement text.
        /// </summary>
        public string Agreement { get; set; }

        /// <summary>
        /// Gets or sets the agreement url.
        /// </summary>
        public string AgreementUrl { get; set; }
    }
}